snapcraft/plugins/python3.py

```diff
@@ -166,10 +167,22 @@ def build(self):
              '--prefix={}/usr'.format(self.installdir),
              '--root={}'.format(self.installdir)], cwd=self.builddir)
 
+        def remove_func(dirpath, files):
+            print(dirpath, files)
+            endswith_pyc = (x.endswith('.pyc') for x in files)
+            return dirpath.endswith('__pycache__') and all(endswith_pyc)
```

Comment:
It would be safer to use os.path.basename(dirpath) == '__pycache__'.
Not that this works only if dirpath doesn't end with a slash, but that's the same assumption in your code.